aiida/cmdline/commands/code.py

```diff
@@ -109,20 +110,18 @@ def __init__(self):
         """
         super(Code, self).__init__()
         self.valid_subcommands = {
-            'list': (self.cli, self.complete_none),
-            'show': (self.cli, self.complete_none),
-            'setup': (self.cli, self.complete_none),
-            'rename': (self.cli, self.complete_none),
-            'relabel': (self.cli, self.complete_none),
-            'update': (self.cli, self.complete_none),
-            'delete': (self.cli, self.complete_none),
-            'hide': (self.cli, self.complete_none),
-            'reveal': (self.cli, self.complete_none),
+            'list': (verdi, self.complete_none),
+            'show': (verdi, self.complete_none),
+            'setup': (verdi, self.complete_none),
+            'rename': (verdi, self.complete_none),
+            'duplicate': (verdi, self.complete_none),
+            'relabel': (verdi, self.complete_none),
+            'update': (verdi, self.complete_none),
+            'delete': (verdi, self.complete_none),
+            'hide': (verdi, self.complete_none),
+            'reveal': (verdi, self.complete_none),
         }
 
-    def cli(self, *args):  # pylint: disable=unused-argument,no-self-use
-        verdi.main()
-
     # pylint: disable=fixme
```

Comment:
Never use this, instead write issues for old TODOs unless they obviously make no sense.